api/src/main/java/org/openmrs/util/OpenmrsUtil.java

```diff
@@ -1099,28 +1094,24 @@ public static String getApplicationDataDirectory() {
 		
 		if (filepath == null) {
 			if (OpenmrsConstants.UNIX_BASED_OPERATING_SYSTEM) {
-				filepath = System.getProperty("user.home") + File.separator + "." + openmrsDir;
+				filepath = Paths.get(System.getProperty("user.home"), ".", openmrsDir).toString();
 				if (!canWrite(new File(filepath))) {
 					log.warn("Unable to write to users home dir, fallback to: "
 					        + OpenmrsConstants.APPLICATION_DATA_DIRECTORY_FALLBACK_UNIX);
-					filepath = OpenmrsConstants.APPLICATION_DATA_DIRECTORY_FALLBACK_UNIX + File.separator + openmrsDir;
+					filepath = Paths.get(OpenmrsConstants.APPLICATION_DATA_DIRECTORY_FALLBACK_UNIX, openmrsDir).toString();
 				}
 			} else {
-				filepath = System.getProperty("user.home") + File.separator + "Application Data" + File.separator + "OpenMRS";
-				if (!new File(filepath).exists()) {
-					filepath = System.getenv("appdata") + File.separator + "OpenMRS";
-				}
```

Comment:
We were reviewing some of the changes we made and were confused on our removal of this if statement. Did we remove this erroneously?

Reply:
Good catch! Yes please keep the if in 👌🏻